pkg/internal/testhelpers/containers.go

```diff
@@ -209,39 +441,120 @@ func StartPGContainerWithImage(ctx context.Context, image string, testDataDir st
 
 	err = container.Start(context.Background())
 	if err != nil {
-		return nil, err
+		return nil, nil, err
+	}
+
+	closer = func() {
+		if printLogs {
+			_ = container.StopLogProducer()
+		}
+
+		_ = container.Terminate(ctx)
+	}
+
+	if isDataNode {
+		err = setDataNodePermissions(container)
+		if err != nil {
+			return nil, closer, err
+		}
 	}
 
 	pgHost, err = container.Host(ctx)
 	if err != nil {
-		return nil, err
+		return nil, closer, err
 	}
 
 	pgPort, err = container.MappedPort(ctx, containerPort)
 	if err != nil {
-		return nil, err
+		return nil, closer, err
 	}
 
 	db, err := pgx.Connect(context.Background(), PgConnectURL(defaultDB, Superuser))
 	if err != nil {
-		return nil, err
+		return nil, closer, err
 	}
 
 	_, err = db.Exec(context.Background(), fmt.Sprintf("CREATE USER %s WITH NOSUPERUSER CREATEROLE PASSWORD 'password'", promUser))
 	if err != nil {
 		//ignore duplicate errors
 		pgErr, ok := err.(*pgconn.PgError)
 		if !ok || pgErr.Code != "42710" {
-			return nil, err
+			return nil, closer, err
+		}
+	}
+
+	if isDataNode {
+		// reload the pg_hba.conf settings we changed in setDataNodePermissions
+		_, err = db.Exec(context.Background(), "SELECT pg_reload_conf()")
+		if err != nil {
+			return nil, closer, err
 		}
 	}
 
 	err = db.Close(context.Background())
 	if err != nil {
-		return nil, err
+		return nil, closer, err
 	}
 
-	return container, nil
+	return container, closer, nil
+}
+
+func enableDataNodes(db *pgx.Conn, database string) error {
```

Comment:
```suggestion
func attachDataNodesOnAN(db *pgx.Conn, database string) error {
```

Reply:
if we're going that direction I think `attachDataNode()`; I think it should be clear we're attaching to the db we're passing in